kyo-core/shared/src/main/scala/kyo/ios.scala

```diff
@@ -118,34 +86,25 @@ sealed trait IOs extends Effect[IOs]:
     def ensure[T, S](f: => Unit < IOs)(v: T < S): T < (IOs & S) =
         val ensure = new Ensure:
             def run = f
-        def ensureLoop(v: T < (IOs & S), p: Preempt): T < (IOs & S) =
-            v match
-                case kyo: Suspend[MX, Any, T, S & IOs] @unchecked =>
-                    new Continue[MX, Any, T, S & IOs](kyo):
-                        def apply(v: Any < (S & IOs), s: Safepoint[S & IOs], l: Locals.State) =
-                            val np =
-                                (s: Any) match
-                                    case s: Preempt =>
-                                        s.ensure(ensure)
-                                        s
-                                    case _ =>
-                                        Preempt.never
-                            val v2 =
-                                try kyo(v, s, l)
-                                catch
-                                    case ex if (NonFatal(ex)) =>
-                                        ensure()
-                                        throw ex
-                            ensureLoop(v2, np)
-                        end apply
-                case _ =>
-                    p.remove(ensure)
-                    IOs {
+        eval(v: T < (IOs & S))(
```

Comment:
This is the most complete example of the usage of the API. The `suspend` parameter can be used to suspend the evaluation with an effect. In this case, both the execution of `resume` and `done` will be wrapped into `IOs` since they perform side effects.